util/src/main/java/com/redhat/lightblue/util/metrics/DropwizardRequestMetrics.java

```diff
@@ -0,0 +1,145 @@
+        return name(metricNamespace, "requests", "exception", actualExceptionClass.getSimpleName(), errorMessage);
+    }
+
+    /**
+     * Get to the cause we actually care about in case the bubbled up exception is a
+     * higher level framework exception that encapsulates the stuff we really care
+     * about.
+     * 
+     */
+    private static Class<? extends Throwable> unravelReflectionExceptions(Throwable e) {
+        while (e.getCause() != null
+                && (e instanceof UndeclaredThrowableException || e instanceof InvocationTargetException)) {
+            e = e.getCause();
+        }
+        return e.getClass();
+    }
+
+    @Override
+    public Context startEntityRequest(String operation, String entity, String version) {
+        return new DropwizardContext(name(API, operation, entity, version));
+    }
+
+    @Override
+    public Context startStreamingEntityRequest(String operation, String entity, String version) {
+        return new DropwizardContext(name(API, "stream", operation, entity, version));
+    }
+    
+    @Override
+    public Context startLockRequest(String lockOperation, String domain) {
+        return new DropwizardContext(name(API, "lock", domain, lockOperation));
+    }
+
+    @Override
+    public Context startSavedSearchRequest(String searchName, String entity, String version) {
+        return new DropwizardContext(name(API, "savedsearch", searchName, entity, version));
+    }
+    
+    @Override
+    public Context startBulkRequest() {
+        return new DropwizardContext(name(API, "bulk"));
+    }
+
+    public class DropwizardContext implements Context {
+        
+        private final String metricNamespace;
+        private final Timer.Context context;
+        private final Counter activeRequests;
+        private boolean ended = false;
+
+        public DropwizardContext(String metricNamespace) {
+            this.metricNamespace = Objects.requireNonNull(metricNamespace, "metricNamespace");
+            this.context = metricsRegistry.timer(name(metricNamespace, "requests", "latency")).time();
+            this.activeRequests = metricsRegistry.counter(name(metricNamespace, "requests", "active"));
+
+            activeRequests.inc();
+        }
+
+        @Override
+        public void endRequestMonitoring() {
+            if (!ended) {
+                ended = true;
+                activeRequests.dec();
+                context.stop();
+            } else {
+                LOGGER.warn("Request already ended for :: " + metricNamespace);
+            }
+        }
+
+        @Override
+        public void markRequestException(Error e) {
+            String errorCode = e.getErrorCode().replaceAll(REGEX, ".");
+            metricsRegistry.meter(errorNamespace(metricNamespace, e, errorCode)).mark();            
+        }
+        
+        @Override
+        public void markRequestException(Exception e) {
+        	String errorMessage = e.getMessage().replaceAll(REGEX, ".");
```

Comment:
Using the error message in the metric could get really messy. These are arbitrary strings, and could be very long. For Error, we're using the error code–a //very// different thing, since these are well known, and short.

Reply:
Removed error message from metric namespace in case of Exception.